apps/meteor/server/routes/avatar/user.ts

```diff
@@ -40,7 +40,20 @@ export const userAvatarByUsername = async function (request: IncomingMessage, re
 
 	// if request starts with @ always return the svg letters
 	if (requestUsername[0] === '@') {
-		serveSvgAvatarInRequestedFormat({ nameOrUsername: requestUsername.slice(1), req, res });
+		const usernameWithoutAt = requestUsername.slice(1);
+		if (settings.get('UI_Use_Name_Avatar')) {
+			const user = await Users.findOneByUsernameIgnoringCase(usernameWithoutAt, {
+				projection: {
+					name: 1,
+				},
+			});
+
+			if (user?.name) {
+				serveSvgAvatarInRequestedFormat({ nameOrUsername: user.name, req, res, useAllInitials: true });
+				return;
+			}
+		}
+		serveSvgAvatarInRequestedFormat({ nameOrUsername: usernameWithoutAt, req, res });
```

Comment:
Unsure if this is the correct place for this logic...

First, if I'm changing my name in my user profile, that won't be reflected in the preview avatar until I save the changes. Secondly, this was never suportted like this, and I don't think it is the resposibility of this resource to deal with this in specific.

I'd argue this should be fixed in the UI (we can just send the name in the place of the username), and for this to also be a separate task, since it's a different issue, also requiring it's own changelog. Can we revert this?